File: Javascript/283.move_zeroes.js
// Given an array nums, write a function to move all 
// 0's to the end of it while maintaining the relative 
// order of the non-zero elements.

var moveZeroes = function (nums) {
    var count = 0

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            nums[count] = nums[i];
            count++;
        }
    }

    for (;count < nums.length; count++) {
        nums[count] = 0;
    }
    return nums;
};

console.log(moveZeroes([0,1,0,3,12]))

File: Ruby/283.move_zeroes.rb
# Given an array nums, write a function to move all 0
# 's to the end of it while maintaining the 
# relative order of the non-zero elements.
def move_zeroes(nums)
    count = 0
    nums.each.with_index do |num, i|
        if num != 0
            nums[count] = num
            count += 1
        end
    end

    while count < nums.length
        nums[count] = 0
        count += 1
    end

    return nums
end

puts move_zeroes([0,1,0,3,12])
